_problems/fee-bumping.md

```diff
@@ -0,0 +1,96 @@
+---
+layout: problem
+title:  "Replacement rules and Fee Bumping"
+tags: lightning
+status: open
+maintainer: LLFourn
+---
+
+Layer-2 protocols like [Lightning] treat the blockchain as a dispute resolution and settlement layer.
+Only when their counterparty is unreachable or uncooperative does a participant have to unilaterally broadcast a transaction.
+The protocol often requires a party to confirm a transaction within a certain time so the funds are distributed fairly.
+So how can layer-2 protocol designers guarantee their users are able to confirm their transaction before the deadline?
```

Comment:
I think we should reevaluate this point. I don't think as a L2 dev you can require "confirmation guarantees" from a distributed system such as Bitcoin. The best you can do is increasing the odds of confirmation by offering an accurate feerate bid for your transactions.

That said as of today, we're even devoid of safe fee-bumping primitives matching L2 protocols semantics. We lack basic security mechanisms but even implemented their efficiency is still function of global blockspace demand.

Ideally, we would define the security properties of L2 protocols through a sequence of security games :
* a zero-sum two-participants game with incomplete information, with only the Lightning holder and its counterparty. Corresponding security property : tx-relay jamming safety (pinning, standardness malleability, witness malleability, ...)
* a zero-sum three-participants game with incomplete information, with the Lightning holder, its counterparty and third-party blockspace bidders. Corresponding security property : congestion safety (feerate reserve)
* a zero-sum four-participants game with incomplete information, with the Lightning holder, its counterparty, third-party blockspace bidders and the miners. Corresponding security property : miner-manipulation resistance
* DLC bonus point : a zero-sum five-participants game with incomplete information, with the Lightning holder, its counterparty, third-party blockspace bidders, the miners, the oracles. Corresponding security property : ?

Of far such security analysis could rely on notions from current framework on cryptographic protocols like the UC model ? That's a good question.

Reply:
Well lightning should come with "security guarantees" under certain assumptions. Your efforts and the goal of this problem are to make these assumptions clear, simple and have a reasonable chance of being realized in practice. This will never be to the same degree as cryptographic hardness assumptions but I still think it's fair to talk about guarantees in this sense for l2 protocols.

I think your assessment of some likely security games with which to define these assumptions in is useful. I'll add a link in the commentary section.

WRT to UC I am not a big fan for these types of systems (I prefer security games). The reason being is that UC is suited to secure computation -- stuff with secret inputs and outputs. Whenever people stray out of this it tends to be really hard to get an intuition for security (for me at least). When you are interacting with a chain you don't really have meaningful secret input. In terms of security you are just trying to make sure that you don't end up in an unacceptable state. To do this you can simply rule out certain types of bad states with a proof by contradiction (and hope you thought of all types of bad states!). Proof assistants can probably help you find all bad states if you have language where you can express l2 protocols easily. It's probably an interesting research topic and deserves a page!